### PR TITLE
Adds mapping freedom to the TEG

### LINKED ISSUE #11075: Feature request for mapping: Make the hardcoded hot/cold circulator locations of a TEG into variables
The locations of the TEG hot and cold circulators are hardcoded. The western one will always be a cold loop and the eastern one a hot loop. Obviously, this greatly limits the way you can design TEG engines when mapping. I'd suggest adding two vars for each circulator so mappers can choose where to place the circulators.

### PR DESCRIPTION
**Pull request by duncathan**
Fixes #11075 
New vars `hot_dir` and `cold_dir` allow mappers to decide which circulator is the hot loop and which is the cold

### CODE REVIEW

File: code/ATMOSPHERICS/components/binary_devices/circulator.dm [hunk 1]
```diff
@@ -15,6 +15,9 @@
 	anchored = 1
 	density = 1
 
+	var/global/const/CIRC_COLD = 1
```
Comment (RemieRichards) [inline code/ATMOSPHERICS/components/binary_devices/circulator.dm]: ```/static```

File: code/modules/power/generator.dm [hunk 2]
```diff
@@ -24,22 +27,19 @@
...
 	connect_to_network()
 
-	if(circ1)
-		circ1.side = 1
-		circ1.update_icon()
-	if(circ2)
-		circ2.side = 2
-		circ2.update_icon()
+	if(cold_circ)
+		cold_circ.side = circpath.CIRC_COLD
```
Comment (RemieRichards) [inline code/modules/power/generator.dm]: This runtimes, you need initial(circpath.CIRC_COLD)

File: code/modules/power/generator.dm [hunk 3]
```diff
@@ -24,22 +27,19 @@
...
+		cold_circ.update_icon()
+	if(hot_circ)
+		hot_circ.side = circpath.CIRC_HOT
```
Comment (RemieRichards) [inline code/modules/power/generator.dm]: This runtimes, you need initial(circpath.CIRC_HOT)

Comment (duncathan) [inline code/modules/power/generator.dm]: what's odd is that that doesn't compile

File: code/ATMOSPHERICS/components/binary_devices/circulator.dm (hunk 1 above)
Comment (duncathan) [inline code/ATMOSPHERICS/components/binary_devices/circulator.dm]: we've had this argument before and I really don't wanna have it again
both are used throughout the codebase and obviously there's no maintainer consensus on it so I'm just gonna leave it as it is

File: code/modules/power/generator.dm (hunk 2 above)
Comment (duncathan) [inline code/modules/power/generator.dm]: fuckin byond with its undocumented bullshit

**Pull request merged**